searx/webapp.py

```diff
@@ -622,12 +666,14 @@ def index():
     )
 
 
-@app.route('/search', methods=['GET', 'POST'])
-def search():
+@app.route('/search', endpoint='search', methods=['GET', 'POST'])
+def route_search():
```

Comment:
I guess the purpose is to avoid name conflict?
What about moving functions that are not routes in some other modules like in https://github.com/searx/searx/pull/2313 ?

Reply:
> I guess the purpose is to avoid name conflict?

yes

> What about moving functions that are not routes in some other modules

I think we should progress modularization more and more in the future, but here it would not help for names like

    @app.route('/preferences', endpoint='preferences', methods=['GET', 'POST'])
    def route_preferences():

Reply:
I prefer to keep the straightforward names:
* The full name is `searx.webapp.preferences`.
* we can rename this to `searx.routes.preferences`, as it is done in [whoogle](https://github.com/benbusby/whoogle-search/blob/main/app/routes.py), so this is explicite --> webapp.py would be the launcher.
* all Flask application I've seen use `endpoint` only as a last resort.

Reply:
> I prefer to keep the straightforward names:
> * The full name is `searx.webapp.preferences`.

The endpoint name is the name of the function `preferences`.

> * we can rename this to `searx.routes.preferences`, as it is done in [whoogle](https://github.com/benbusby/whoogle-search/blob/main/app/routes.py), so this is explicite --> webapp.py would be the launcher.

Yes, on a long term we should modularise into a module named `routes` .. but the endpoint name will be the name of the function.

> * all Flask application I've seen use `endpoint` only as a last resort.

They have clean namespaces and modularisation

This PR is dedicated to linit and modularisation is not in the scope .. we can decide this later.  I add a `# pylint: disable=redefined-outer-name` to the lines causing name clashes.

Reply:
> but the endpoint name will be the name of the function.

does it cause any sort of problem?

Reply:
I am not familiar in deep but for flask itself endpoints are substantial for the routing.

- see section  [Endpoints and URLs](https://flask.palletsprojects.com/en/2.0.x/tutorial/views/#endpoints-and-urls)
- https://flask.palletsprojects.com/en/2.0.x/api/#flask.Flask.add_url_rule

We use it here (calls are coming from the templates):

https://github.com/searxng/searxng/blob/b1da97d0922020dabcdc971d16e3e7a95be58c1b/searx/webapp.py#L346-L353

And we use this:

https://github.com/searxng/searxng/blob/b1da97d0922020dabcdc971d16e3e7a95be58c1b/searx/webapp.py#L473

to set the `<body class="{{ endpoint }}_endpoint" >` in the templates.